### PR TITLE
Adds a set password page

### PR DESCRIPTION
**Pull request by tgolen**
### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/148712

### Tests
1. Sign out
2. Go to http://localhost:8080/#/setpassword/test
3. Submit the form without a password and verify there is an error the password is required
4. Enter a password and submit the form
5. Verify that (for now) you are just taken back to the sign in page

### Screenshots
image

### CODE REVIEW

File: src/pages/SignInPage.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ class App extends Component {
             <>
                 <CustomStatusBar />
                 <SafeAreaView style={[styles.signInPage]}>
-                    <View style={[styles.signInPageInner]}>
+                    <View style={[styles.signInPageInner]} accessibilityRole="form">
```
Comment (tgolen) [inline src/pages/SignInPage.js:89]: FYI this removes a warning in the JS console about a password field not being inside of a form.

**Comment by tgolen** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/pages/SetPasswordPage.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    /* Onyx Props */
+
+    // The credentials of the logged in person
+    credentials: PropTypes.shape({
+        // The email the user logged in with
+        login: PropTypes.string,
+
+        // The password used to log in the user
+        password: PropTypes.bool,
```
Comment (TomatoToaster) [inline src/pages/SetPasswordPage.js:38]: I think this should be type string.

**Review by TomatoToaster — CHANGES_REQUESTED** [review]

File: src/pages/SetPasswordPage.js (hunk 2 above)
Comment (tgolen) [reply src/pages/SetPasswordPage.js:38]: Oh yes indeed!

**Comment by tgolen** [comment]
Updated

**Review by TomatoToaster — DISMISSED** [review]

**Comment by TomatoToaster** [comment]
Do you know how I can clear the credentials object so I can try that page again. I could've sworn that it wouldn't let me submit the password after the error showed up even if I entered something non-blank (had to refresh the page).

**Comment by tgolen** [comment]
Yeah, you can to JS console > Application > Local Storage > localhost (right click) > clear

**Comment by tgolen** [comment]
I was able to get it to submit the form even after entering no password, but I see that it doesn't clear the error. I will fix that.

**Comment by tgolen** [comment]
Updated

**Review by TomatoToaster — APPROVED** [review]

**Review by stitesExpensify — APPROVED** [review]

**Pull request merged**